### PR TITLE
Fix a bug in calculating the effective number of selected indices in NonMaxSuppression

### LINKED ISSUE #1158: NonMaxSuppresion in ssd-10 model produced a wrong result.
Inference accuracy of [ssd-10](https://github.com/onnx/models/blob/master/vision/object_detection_segmentation/ssd/README.md) was wrong.

I used RunONNXModel.py to check which op caused the wrong result.
```
$ ONNX_MLIR_HOME=/Users/tung/dl/onnx-mlir/build/Debug python ../utils/RunONNXModel.py ~/Downloads/ssd-10/model.onnx --compile_args="-O3" --verify=onnxruntime --data_folder=/Users/tung/Downloads/ssd-10/test_data_set_0/ --save_onnx=./ssd10-verify.onnx --verify_all_ops 2>&1 | tee verify-ssd10.log
```
and found that it is NonMaxSuppresion:
```
Verifying value of NonMaxSuppression_683:[599, 3] using atol=0.01, rtol=0.05 ...
  at (110, 0) mismatch -1 (actual) vs 0 (reference)                                                                                                              at (110, 1) mismatch -1 (actual) vs 2 (reference)                                                                                                              at (110, 2) mismatch -1 (actual) vs 10977 (reference)
  at (111, 0) mismatch -1 (actual) vs 0 (reference)                                                                                                              at (111, 1) mismatch -1 (actual) vs 2 (reference)                                                                                                              at (111, 2) mismatch -1 (actual) vs 3904 (reference)
  at (112, 0) mismatch -1 (actual) vs 0 (reference)                                                                                                              at (112, 1) mismatch -1 (actual) vs 2 (reference)                                                                                                              at (112, 2) mismatch -1 (actual) vs 14011 (reference)
  at (113, 0) mismatch -1 (actual) vs 0 (reference)                                                                                                              at (113, 1) mismatch -1 (actual) vs 2 (reference)                                                                                                              at (113, 2) mismatch -1 (actual) vs 1402 (reference)
  at (114, 0) mismatch -1 (actual) vs 0 (reference)                                                                                                              at (114, 1) mismatch -1 (actual) vs 2 (reference)                                                                                                              at (114, 2) mismatch -1 (actual) vs 6350 (reference)
  at (115, 0) mismatch -1 (actual) vs 0 (reference)                                                                                                              at (115, 1) mismatch -1 (actual) vs 2 (reference)                                                                                                              at (115, 2) mismatch -1 (actual) vs 1351 (reference)
  at (116, 0) mismatch -1 (actual) vs 0 (reference)
```

This is NonMaxSuppresion  in ssd-10:

```mlir
%487 = "onnx.Concat"(%483, %484, %485, %486) {axis = 2 : si64, onnx_node_name = "Concat_659"} : (tensor<?x15130x1xf32>, tensor<?x15130x1xf32>, tensor<?x15130x1xf32>, tensor<?x15130x1xf32>) -> tensor<?x15130x4xf32>
%504 = "onnx.Slice"(%499, %500, %501, %502, %503) {onnx_node_name = "Slice_676"} : (tensor<?x?x?xf32>, tensor<1xi64>, tensor<1xi64>, tensor<1xi64>, tensor<1xi64>) -> tensor<?x?x?xf32>
%506 = "onnx.ConstantOfShape"(%505) {onnx_node_name = "ConstantOfShape_678", value = dense<200> : tensor<1xi64>} : (tensor<1xi64>) -> tensor<1xi64>
%508 = "onnx.ConstantOfShape"(%507) {onnx_node_name = "ConstantOfShape_680", value = dense<5.000000e-01> : tensor<1xf32>} : (tensor<1xi64>) -> tensor<1xf32>
%510 = "onnx.ConstantOfShape"(%509) {onnx_node_name = "ConstantOfShape_682", value = dense<5.000000e-02> : tensor<1xf32>} : (tensor<1xi64>) -> tensor<1xf32>
 %511 = "onnx.NonMaxSuppression"(%487, %504, %506, %508, %510) {onnx_node_name = "NonMaxSuppression_683"} : (tensor<?x15130x4xf32>, tensor<?x?x?xf32>, tensor<1xi64>, tensor<1xf32>, tensor<1xf32>) -> tensor<?x3xi64>
```

### PR DESCRIPTION
**Pull request by tungld**
Resolves #1158 

Signed-off-by: Tung D. Le <tungld@gmail.com>

### CODE REVIEW

File: src/Conversion/ONNXToKrnl/ObjectDetection/NonMaxSuppression.cpp [hunk 1]
```diff
@@ -423,7 +420,7 @@ struct ONNXNonMaxSuppressionOpLowering : public ConversionPattern {
                           createMath, selectedBox, otherBox, centerPointBox);
 
                       // Only proceed if IOU > iou_threshold.
```
Comment (AlexandreEichenberger) [inline src/Conversion/ONNXToKrnl/ObjectDetection/NonMaxSuppression.cpp:422]: nit, does the comment needs to be update to `>=` ?

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM

File: src/Conversion/ONNXToKrnl/ObjectDetection/NonMaxSuppression.cpp (hunk 1 above)
Comment (tungld) [reply src/Conversion/ONNXToKrnl/ObjectDetection/NonMaxSuppression.cpp]: Yes, just updated it. Thanks!

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by gongsu832** [comment]
@jenkins-droid test this please

**Comment by gongsu832** [comment]
@jenkins-droid test this please

**Comment by gongsu832** [comment]
Something related to build script appears to be wrong with this particular PR. I'm looking into it. Could we please not touch this PR until I find the problem? Thanks.

**Comment by gongsu832** [comment]
@jenkins-droid test this please

**Comment by gongsu832** [comment]
OK the problem is that somehow a `\r` got into the commit message (we normally only get `\n`). https://github.com/onnx/onnx-mlir/pull/1162 should fix it.

**Comment by tungld** [comment]
> OK the problem is that somehow a \r got into the commit message (we normally only get \n). https://github.com/onnx/onnx-mlir/pull/1162 should fix it.

Thank you, Gong, for fixing the issue! The last commit was submitted via web GUI not command-line, maybe it appended \r.

**Pull request merged**